junit-vintage-engine/src/main/java/org/junit/vintage/engine/VintageTestEngine.java

```diff
@@ -92,32 +95,46 @@ public void execute(ExecutionRequest request) {
 
 	private void executeAllChildren(VintageEngineDescriptor engineDescriptor,
 			EngineExecutionListener engineExecutionListener, ExecutionRequest request) {
-		boolean parallelExecutionEnabled = getParallelExecutionEnabled(request);
+		initializeParallelExecutionParameters(request);
 
-		if (parallelExecutionEnabled) {
-			if (executeInParallel(engineDescriptor, engineExecutionListener, request)) {
-				Thread.currentThread().interrupt();
-			}
+		boolean parallelExecutionEnabled = getParallelExecutionEnabled(request);
+		if (!parallelExecutionEnabled) {
+			executeSequentially(engineDescriptor, engineExecutionListener);
+			return;
 		}
-		else {
+
+		if (!classes && !methods) {
+			logger.warn(() -> "Parallel execution is enabled but no scope is defined. "
+					+ "Falling back to sequential execution.");
 			executeSequentially(engineDescriptor, engineExecutionListener);
+			return;
+		}
+
+		if (executeInParallel(engineDescriptor, engineExecutionListener, request)) {
```

Comment:
I think it's more readable to give the returned `boolean` a name

```suggestion
		boolean wasInterrupted = executeInParallel(engineDescriptor, engineExecutionListener, request);
		if (wasInterrupted) {
```